tests/test_write_to_catalog.py

```diff
@@ -0,0 +1,139 @@
+from copy import deepcopy
+from json import load
+
+from mypy_boto3_s3 import S3Client
+from pytest import mark
+from pytest_subtests import SubTests  # type: ignore[import]
+from smart_open import smart_open  # type: ignore[import]
+
+from backend.resources import ResourceName
+from backend.stac_format import (
+    STAC_CHILD_KEY,
+    STAC_DESCRIPTION_KEY,
+    STAC_HREF_KEY,
+    STAC_ID_KEY,
+    STAC_LINKS_KEY,
+    STAC_REL_KEY,
+    STAC_ROOT_KEY,
+    STAC_TITLE_KEY,
+    STAC_TYPE_KEY,
+)
+from backend.types import JsonList
+from backend.write_to_catalog import task
+from backend.write_to_catalog.task import (
+    CATALOG_KEY,
+    ROOT_CATALOG_DESCRIPTION,
+    ROOT_CATALOG_ID,
+    ROOT_CATALOG_TITLE,
+)
+from tests.aws_utils import Dataset, S3Object, any_lambda_context, delete_s3_key
+from tests.file_utils import json_dict_to_file_object
+from tests.stac_objects import MINIMAL_VALID_STAC_CATALOG_OBJECT
+
+
+@mark.infrastructure
+def should_create_new_root_catalog_if_doesnt_exist(subtests: SubTests, s3_client: S3Client) -> None:
+
+    with Dataset() as dataset, S3Object(
+        file_object=json_dict_to_file_object(
+            {
+                **deepcopy(MINIMAL_VALID_STAC_CATALOG_OBJECT),
+                STAC_ID_KEY: dataset.dataset_id,
+                STAC_TITLE_KEY: dataset.title,
+            }
+        ),
+        bucket_name=ResourceName.STORAGE_BUCKET_NAME.value,
+        key=f"{dataset.dataset_prefix}/{CATALOG_KEY}",
+    ):
+
+        body = {"Records": [{"body": dataset.dataset_prefix}]}
+
+        try:
+            task.lambda_handler(body, any_lambda_context())
+
+            expected_links: JsonList = [
+                {
+                    STAC_REL_KEY: STAC_ROOT_KEY,
+                    STAC_HREF_KEY: f"./{CATALOG_KEY}",
```

Comment:
Is it possible to avoid including the "./" part at the start of these `href`s? I'm not sure it's valid in S3, and it's redundant.